### PR TITLE
Simplify handling of --no-interaction and confirmation

### PR DESCRIPTION
**Pull request by rjmunro**
If no interaction is set, pre-set confirmation to true, then only call execute from one place.

Reduces the number of places you have to update the code if you want to add more parameters to the execute method, e.g. --ignore-errors (https://github.com/doctrine/migrations/pull/188)

### CODE REVIEW

File: lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php [hunk 1]
```diff
@@ -84,14 +84,14 @@ public function execute(InputInterface $input, OutputInterface $output)
         } else {
             $noInteraction = $input->getOption('no-interaction') ? true : false;
             if ($noInteraction === true) {
-                $version->execute($direction, $input->getOption('dry-run') ? true : false);
+                $confirmation = true;
```
Comment (mikeSimonson) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php]: Maybe changing the name of the variable to $execute make it a bit more understandable ?

File: lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php [hunk 2]
```diff
@@ -84,14 +84,14 @@ public function execute(InputInterface $input, OutputInterface $output)
...
             } else {
                 $confirmation = $this->getHelper('dialog')->askConfirmation($output, '<question>WARNING! You are about to execute a database migration that could result in schema changes and data lost. Are you sure you wish to continue? (y/n)</question>', false);
-                if ($confirmation === true) {
-                    $version->execute($direction, $input->getOption('dry-run') ? true : false);
-                } else {
-                    $output->writeln('<error>Migration cancelled!</error>');
-                }
+            }
+            if ($confirmation === true) {
+                $version->execute($direction, $input->getOption('dry-run') ? true : false);
```
Comment (stof) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php]: can you remove the ``? true : false`` ternary operator too ? It makes no sense to turn a boolean into another boolean.

Comment (mikeSimonson) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php]: Isn't getOption a return mixed function ?

Comment (stof) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php]: it depends how the option is configured. For option without value (which is the case here), it always return a boolean. For options with a value, it returns a string (the CLI only allows to pass strings)

File: lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php [hunk 3]
```diff
@@ -82,16 +82,16 @@ public function execute(InputInterface $input, OutputInterface $output)
             $path = is_bool($path) ? getcwd() : $path;
             $version->writeSqlFile($path, $direction);
         } else {
-            $noInteraction = $input->getOption('no-interaction') ? true : false;
-            if ($noInteraction === true) {
-                $version->execute($direction, $input->getOption('dry-run') ? true : false);
+            if ($input->getOption('no-interaction')) {
```
Comment (stof) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php]: ``$input->isInteractive()`` would be even cleaner actually (the ``no-interaction`` option is an Application-level option, but the Application has other ways to guess that the input cannot be interactive in some case).

Comment (rjmunro) [inline lib/Doctrine/DBAL/Migrations/Tools/Console/Command/ExecuteCommand.php]: I'm a bit nervous as this changes functionality slightly - I've seen people use constructs like `echo y | app/console doctrine:migrations:migrate` because they didn't know about the `--no-interaction` option - that kind of thing might break. I'd rather put that in a separate PR from this, which is just intended to tidy some things a bit.

**Comment by rjmunro** [comment]
@stof Is this ready to be merged now, or do I need to do anything else?

**Comment by mikeSimonson** [comment]
@rjmunro I am just waiting for the input of others on other PR before to merge this one but I feel likes it's ready to be merged.

**Comment by mikeSimonson** [comment]
@rjmunro Thanks
